### ISSUE TITLE
MxNet Pre-trained Prediction Doc Not Working

### ISSUE THREAD

**Issue by prathik-naidu** [apache/incubator-mxnet#12063]
## Description
The predict image doc page: [https://mxnet.incubator.apache.org/tutorials/python/predict_image.html](url) doesn't work right off the shelf (returns incorrect predictions).

## Problem
The tutorial doesn't normalize the images – causing incorrect predictions to be made. For newcomers to MxNet/deep learning, I think this is a simple, but useful addition to add in.

## Solution
Add in a color normalization in the get_image method as follows:

```python
def get_image(url, show=False):
    # download and show the image. Remove query string from the file name.
    fname = mx.test_utils.download(url, fname=url.split('/')[-1].split('?')[0])
    img = mx.image.imread(fname)
    if img is None:
        return None
    if show:
        plt.imshow(img.asnumpy())
        plt.axis('off')
    # convert into format (batch, RGB, width, height)
    img = mx.image.imresize(img, 224, 224) # resize
    #---------------------------
    img = img.astype(float)/255
    img = mx.image.color_normalize(img,
               mean=mx.nd.array([0.485, 0.456, 0.406]).astype(float), 
               std=mx.nd.array([0.229, 0.224, 0.225]).astype(float)
    #---------------------------
    img = img.transpose((2, 0, 1)) # Channel first
    img = img.expand_dims(axis=0) # batchify
    return img
```

**Comment by marcoabreu** [comment]
@mxnet-label-bot Example Bug

**Comment by Ishitori** [comment]
@prathik-naidu I think that model may actually have been trained without normalization. 

I receive worse results if I apply normalization to the data. The top one is: `probability=0.060943, class=n01930112 nematode, nematode worm, roundworm` which is clearly worse than original one: `probability=0.249607, class=n02119022 red fox, Vulpes vulpes`

If I change model to Gluon one (`resnet18 = vision.resnet18_v1(pretrained=True)`), then it is clear that it was trained with normalization in place. With normalization I get: `probability=8.849780, class=n02124075 Egyptian cat`, without I get: `probability=60.358135, class=n04270147 spatula`

So, I am not sure how this model was trained. It could be that it actually didn't use normalization. What results do you get if you apply normalization?

**Comment by piyushghai** [comment]
@mxnet-label-bot [Pending Requester Info]

**Comment by ankkhedia** [comment]
@prathik-naidu Did you get a chance to look into results after applying normalization?

**Comment by ankkhedia** [comment]
@prathik-naidu I ran the predictions with the mxnet-1.3.0 and see same predictions as mentioned in the example itself. As @sergeykolychev pointed out the predictions get worse with normalisation and maybe the model was trained without normalisation.

**Comment by ankkhedia** [comment]
@prathik-naidu 
This model has been trained without normalization and doesn't need its input normalized, contrary to the Gluon model zoo one that needs its input normalized. 
So closing the issue as the example is working as expected. Please feel free to re-open if closed in error.

@sandeep-krishnamurthy Could you please close the issue
